source/include/boot_upgrade_package.h

```diff
@@ -6,13 +6,17 @@
 
 typedef struct
 {
-    uint32_t dummy;
-    /* TODO all fields here */
+ 
+    uint8_t prefix[4];
+    uint32_t version;
+    uint32_t size;
```

Comment:
Please add a 4 byte reserved variable after the size to locate the signature to the offset 16.

`uint8_t __reserved[4];`

Reply:
Done